kadasrouting/gui/navigationpanel.py

```diff
@@ -319,19 +323,23 @@ def updateNavigationInfo(self):
         layer = self.iface.activeLayer()
         LOG.debug("Debug: type(layer) = {}".format(type(layer)))
         point = QgsPointXY(gpsinfo.longitude, gpsinfo.latitude)
+        if gpsinfo.speed > GPS_MIN_SPEED:
+            # if we are moving, it is better for the user experience to
+            # project the current point using the speed vector instead
+            # of using 'point' directly, otherwise we get to feel of being
+            # "behind the current position"
+            qgsdistance = QgsDistanceArea()
+            qgsdistance.setSourceCrs(
+                QgsCoordinateReferenceSystem(4326),
+                QgsProject.instance().transformContext(),
+            )
+            qgsdistance.setEllipsoid(qgsdistance.sourceCrs().ellipsoidAcronym())
+            point = qgsdistance.computeSpheroidProject(
+                point, (gpsinfo.speed / 2.0) * REFRESH_RATE_S, math.radians(gpsinfo.direction)
```

Comment:
Perhaps we can put this 2.0 as a variable, just like REFRESH_RATE_S and GPS_MIN_SPEED so that we understand what is it (and why) when we need to check it in the future

Reply:
good point, yes